sdk/keyvault/azure-security-keyvault-administration/src/test/java/com/azure/security/keyvault/administration/KeyVaultCredentialPolicyTest.java

```diff
@@ -146,10 +137,53 @@ public void onChallengeCredentialPolicy() {
     public void onAuthorizeRequestChallengeCachePresent() {
         KeyVaultCredentialPolicy policy = new KeyVaultCredentialPolicy(this.credential, false);
 
+        StepVerifier.create(onChallenge(policy, this.callContext, unauthorizedHttpResponseWithHeader) // Challenge cache created
+                .then(policy.authorizeRequest(this.testContext))) // Challenge cache used
+            .verifyComplete();
+
+        String tokenValue = this.testContext.getHttpRequest().getHeaders().getValue(HttpHeaderName.AUTHORIZATION);
+        assertFalse(tokenValue.isEmpty());
+        assertTrue(tokenValue.startsWith(BEARER));
+    }
+
+    @Test
+    public void onAuthorizeRequestChallengeCachePresentSync() {
+        KeyVaultCredentialPolicy policy = new KeyVaultCredentialPolicy(this.credential, false);
+
         // Challenge cache created
-        onChallenge(policy, this.callContext, unauthorizedHttpResponseWithHeader).block();
+        onChallengeSync(policy, this.callContext, unauthorizedHttpResponseWithHeader);
         // Challenge cache used
-        policy.authorizeRequest(this.testContext).block();
+        policy.authorizeRequestSync(this.testContext);
+
+        String tokenValue = this.testContext.getHttpRequest().getHeaders().getValue(HttpHeaderName.AUTHORIZATION);
+        assertFalse(tokenValue.isEmpty());
+        assertTrue(tokenValue.startsWith(BEARER));
+    }
+
+    @Test
+    public void onAuthorizeRequestChallengeCachePresentWithClaims() {
+        KeyVaultCredentialPolicy policy = new KeyVaultCredentialPolicy(this.credential, false);
+
+        StepVerifier.create(onChallenge(policy, this.callContext, unauthorizedHttpResponseWithHeader) // Challenge cache created
+                .then(onChallenge(policy, this.callContext, unauthorizedHttpResponseWithHeader)) // Challenge with claims received
```

Comment:
The comment suggests that this should be a claims challenge, but isn't `unauthorizedHttpResponseWithHeader` a regular KV challenge?

Reply:
You're right, I'll change it to use the one with claims, thanks for catching that!